src/sqlfluff/core/linter/linter.py

```diff
@@ -494,112 +495,142 @@ def lint_fix_parsed(
             ignore_buff = []
 
         save_tree = tree
-        for loop in range(loop_limit):
-            linter_logger.info(f"Linter loop {loop+1}/{loop_limit}")
-            changed = False
-
-            progress_bar_crawler = tqdm(
-                rule_set,
-                desc="lint by rules",
-                leave=False,
-                disable=progress_bar_configuration.disable_progress_bar,
-            )
-
-            for crawler in progress_bar_crawler:
-                progress_bar_crawler.set_description(f"rule {crawler.code}")
-
-                # fixes should be a dict {} with keys edit, delete, create
-                # delete is just a list of segments to delete
-                # edit and create are list of tuples. The first element is the
-                # "anchor", the segment to look for either to edit or to insert BEFORE.
-                # The second is the element to insert or create.
-                linting_errors, _, fixes, _ = crawler.crawl(
-                    tree,
-                    ignore_mask=ignore_buff,
-                    dialect=config.get("dialect_obj"),
-                    fname=fname,
-                    fix=fix,
-                    templated_file=templated_file,
+        # There are two phases of rule running. The main loop is for most rules.
+        # These rules are assumed to interact and cause a cascade of fixes
+        # requiring multiple passes. The post loop is for post-processing rules,
+        # not expected to trigger any downstream rules, e.g. capitalization fixes.
+        phases = ["main"]
+        if fix:
+            phases.append("post")
+        for phase in phases:
+            if len(phases) > 1:
+                rules_this_phase = [
+                    rule for rule in rule_set if rule.lint_phase == phase
+                ]
+            else:
+                rules_this_phase = rule_set
+            for loop in range(loop_limit if phase == "main" else 2):
+                linter_logger.info(f"Linter phase {phase}, loop {loop+1}/{loop_limit}")
+                changed = False
+
+                if phase == "main" and loop == 0:
+                    # In order to compute initial_linting_errors correctly, need
+                    # to run all rules on the first loop of the main phase.
+                    rules_this_phase = rule_set
```

Comment:
Main phase, loop 1 **always runs all the rules**. This guarantees that `initial_linting_errors` includes output from all the rules. Should be identical behavior as before.